### PR TITLE
Support ESLint 8.x

### LINKED ISSUE #62: Support ESLint 8.x
https://eslint.org/blog/2021/10/eslint-v8.0.0-released

- [ ] `@typescript-eslint/eslint-plugin`: Upgrade to https://github.com/typescript-eslint/typescript-eslint/releases/tag/v5.0.0
- [ ] `@typescript-eslint/parser`: Upgrade to https://github.com/typescript-eslint/typescript-eslint/releases/tag/v5.0.0
- [ ] `babel-eslint`: Migrate to https://www.npmjs.com/package/@babel/eslint-parser
- [ ] `babel-eslint`: Upgrade to https://github.com/babel/babel/releases/tag/v7.16.0
- [ ] `eslint-config-prettier`: Upgrade to https://github.com/prettier/eslint-config-prettier/blob/HEAD/CHANGELOG.md#version-800-2021-02-21?
- [ ] `eslint-import-resolver-node`: Validate that this works with latest `eslint-plugin-import`
- [x] `eslint-plugin-css-modules`: #64
- [ ] ⛔ `eslint-plugin-cypress`: https://github.com/cypress-io/eslint-plugin-cypress/issues/89
- [x] `eslint-plugin-flowtype`: #64
- [ ] `eslint-plugin-import`: Upgrade to https://github.com/import-js/eslint-plugin-import/releases/tag/v2.25.0
- [x] `eslint-plugin-jest`: #60 https://github.com/seek-oss/eslint-config-seek/releases/tag/v7.0.9
- [ ] `eslint-plugin-react`: Upgrade to https://github.com/yannickcr/eslint-plugin-react/releases/tag/v7.27.0
- [ ] `eslint-plugin-react-hooks`: Upgrade to https://github.com/facebook/react/pull/22248#issuecomment-964176674

### PR DESCRIPTION
**Pull request by 72636c**
I believe the only remaining incompatibility is with the Cypress plugin. This shouldn't be a blocker as we still support ESLint 7.x in theory.

Closes #62.

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -118,7 +121,7 @@ const baseConfig = {
       extends: [
         'plugin:@typescript-eslint/eslint-recommended',
         'plugin:@typescript-eslint/recommended',
-        'prettier/@typescript-eslint',
+        'prettier',
```
Comment (72636c) [inline .eslintrc.js:124]: Per https://github.com/prettier/eslint-config-prettier/blob/HEAD/CHANGELOG.md#version-800-2021-02-21

File: package.json [hunk 2]
```diff
@@ -19,22 +19,24 @@
...
+    "eslint-plugin-jest": "^26.1.2",
+    "eslint-plugin-react": "^7.29.4",
+    "eslint-plugin-react-hooks": "^4.3.0",
     "find-root": "^1.1.0"
   },
   "devDependencies": {
     "@changesets/cli": "2.17.0",
     "@changesets/get-github-info": "0.5.0",
-    "eslint": "^7.14.0",
+    "eslint": "7.32.0",
```
Comment (mattcompiles) [inline package.json:39]: Shouldn't this be 8 now? Realise it's a dev dep.

**Comment by 72636c** [comment]
Would appreciate eyes on this one @seek-oss/sku-maintainers. This seems to work fine for skuba and a couple sku apps I tested via Yarn resolution override, but you'll have a more holistic view. There's a beta published: https://github.com/seek-oss/eslint-config-seek/releases/tag/v9.0.0-beta.0

The TypeScript ESLint bump should allow us to upgrade to newer TypeScript versions without an annoying warning being printed to the terminal.

**Review by mattcompiles — APPROVED** [review]
All makes sense mate. Was just going to ask if you'd tested it in a real app but sounds like you've got that covered. Seeing as you're versioning as a major I think it's safe to merge. 👍

**Pull request merged**